server-routes/src/prove.rs

```diff
@@ -35,10 +35,21 @@ pub async fn prove(mut req: ProveRequest) -> Result<serde_json::Value, Box<dyn s
         req.abi.this_addr = Some(0);
     }
 
+    let this_salts = req
```

Comment:
We should use the `req.this_salts` if provided, otherwise generate the zeroed default

Reply:
Done.